app/org/maproulette/controllers/api/ChallengeController.scala

```diff
@@ -348,7 +348,22 @@ class ChallengeController @Inject()(override val childController: TaskController
     */
   def deleteTasks(challengeId: Long, statusFilters: String = ""): Action[AnyContent] = Action.async { implicit request =>
     this.sessionManager.authenticatedRequest { implicit user =>
-      this.dal.deleteTasks(user, challengeId, Utils.split(statusFilters).map(_.toInt))
+      val originalStatus = dalManager.challenge.retrieveById(challengeId) match {
+        case Some(c) => c.status
+        case None => throw new NotFoundException(s"No challenge found with id $challengeId")
+      }
+
+      // Deleting a lot of tasks can be time consuming, so perform this asynchronously
+      // Todo: perhaps introduce a DELETING status rather than co-opting BUILDING
+      dalManager.challenge.update(Json.obj("status" -> Challenge.STATUS_BUILDING), user)(challengeId)
```

Comment:
I very much agree with this todo. Any reason why you don't just add that status in this PR?

Reply:
Will do ;-)